views/forms/RegisterForm/index.jsx

```diff
@@ -1,73 +1,91 @@
-import Form from '../Form'
-import TextInput from '../TextInput'
 import { Button } from '@mui/material'
+import { useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+
+import TextInput from '../TextInput'
+
+const schema = yup.object({
+  username: yup.string().required(),
+  password: yup.string().required().min(8),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref('password'), null], 'passwords do not match'),
+  fullName: yup.string().required(),
+  email: yup.string().required().email(),
+})
+
+const RegistrationForm = ({ initialValues, onCancel, onSubmit }) => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: initialValues,
+    resolver: yupResolver(schema),
+  })
 
-const RegistrationForm = ({
-  onSubmit,
-  control,
-  errors,
-  navigate,
-  onInputChange,
-}) => {
   return (
-    <Form onSubmit={onSubmit}>
+    <form onSubmit={handleSubmit(onSubmit)}>
       <TextInput
         control={control}
-        name="username"
+        errors={errors.username}
+        fullWidth
         label="Username"
-        required={true}
         margin="normal"
+        name="username"
+        required={true}
```

Comment:
```suggestion
```
The yup schema has it defined

Reply:
right but this is for the HTML input element. It doesn't hurt to have it in both places, it only helps.

Reply:
I wonder if we would even make it to this because of the yup validation

Reply:
it's the opposite, we wouldn't make it to the yup validation because of this. Unless a user removed the attribute from the DOM.